eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -177,7 +191,30 @@ class EclairImpl(appKit: Kit) extends Eclair {
     case Some(pk) => (appKit.router ? 'updatesMap).mapTo[Map[ChannelDesc, ChannelUpdate]].map(_.filter(e => e._1.a == pk || e._1.b == pk).values)
   }
 
+  override def getNetworkInfoResponse()(implicit timeout: Timeout): Future[GetNetworkInfoResponse] = {
+    for {
+      nodeCount <- allNodes().map{_.size}
+      channelCount <- allChannels().map{_.size}
+
+      (updateCount: Long,avgCltvExpiryDelta: Long,avgHtlcMinimumMsat: Long,avgFeeBaseMsat: Long,avgFeeProportionalMillionths: Long,avgHtlcMaximumMsat: Long, mf: FlagCounter, cf:FlagCounter ) <- allUpdates(None).map{
+        updates => updates.foldLeft(Tuple8(0L, 0L, 0L, 0L, 0L, 0L,FlagCounter(), FlagCounter())) {
+          (t, c) =>
+            Tuple8(t._1 + 1,
```

Comment:
Instead of this hard-to-read `Tuple8` why don't you directly use a `GetNetworkInfoResponse` here?
I think it would be cleaner.

Reply:
Not sure this is possible. As GetNetworkInfoResponse include results of the earlier futures too.
Could fill with dummy zeros but think that would make more confusing.
Especially if additional calls/futures were added to get more data.

Reply:
I would do something like (I slightly renamed the fields, you'll have to adapt):

```
override def getNetworkInfoResponse()(implicit timeout: Timeout): Future[GetNetworkInfoResponse] = {
    for {
      nodeCount <- allNodes().map(_.size)
      channelCount <- allChannels().map(_.size)
      channelUpdates <- allUpdates(None)
      val networkInfo = channelUpdates.foldLeft(GetNetworkInfoResponse(channelCount, nodeCount, channelUpdates.size, 0, MilliSatoshi(0), MilliSatoshi(0), 0, MilliSatoshi(0))) {
        (current, next) => current.copy(
          avgCltvExpiryDelta = current.avgCltvExpiryDelta + next.cltvExpiryDelta,
          avgFeeBaseMsat = current.avgFeeBaseMsat + next.feeBaseMsat
          // etc
        )
      }
    } yield networkInfo.copy(
      avgFeeBaseMsat = networkInfo.avgFeeBaseMsat / networkInfo.updates
      // etc
    )
  }
```

And I would refactor to make dividing all `avg` fields by the `updateCount` be done in a method in the case class or something like that.